### PR TITLE
Nemo/Update Rails

### PR DESCRIPTION
**Pull request by nemodjuric**
Connects to https://github.com/department-of-veterans-affairs/caseflow-feedback/issues/157
Update Rails to version 5

Rollout plan:
- Deploy and test it in UAT before each merge
- Make an announcement about the upgrade and changes introduced from developer's point common tasks point of view after each app upgrade (discuss new_framework_defaults.rb)
Any changes in caseflow gem (caseflow-commons) to be merged should be compatible to older rails versions. Testing these changes should be done locally first

- Upgrade to Rails 5 in `caseflow-feedback`

- Upgrade `monitor` to Rails 5

- Upgrade `caseflow-efolder` to Rails 5

- Upgrade `caseflow` to Rails 5

- Remove old code from `caseflow-commons` and all `# TODO Rails5Upgrade

- Make a final announcement about completion of the task

### CODE REVIEW

File: config/environments/development.rb
Comment (nemodjuric) [inline config/environments/development.rb]: This is set to true by default.

File: config/environments/test.rb
Comment (nemodjuric) [inline config/environments/test.rb]: This is set by default.

File: Gemfile
Comment (nemodjuric) [inline Gemfile]: This caseflow gem version includes autoloading logger compatibility for Rails 5.

File: Gemfile
Comment (nemodjuric) [inline Gemfile]: 5.1.5 is the latest stable Rails version.

File: app/models/admin.rb
Comment (nemodjuric) [inline app/models/admin.rb]: In Rails 5.0, all models inherit from ApplicationRecord.

File: db/schema.rb
Comment (nemodjuric) [inline db/schema.rb]: We omitted B-tree, since it's a default index type for PostgreSQL. (ref. https://www.postgresql.org/docs/9.5/static/indexes-types.html)

File: db/schema.rb
Comment (nemodjuric) [inline db/schema.rb]: This was an automated change by rails update (ref. https://apidock.com/rails/ActiveRecord/ConnectionAdapters/Table/index)

File: spec/feature/admin_page_spec.rb
Comment (nemodjuric) [inline spec/feature/admin_page_spec.rb]: This was an old syntax and it was giving deprecation warnings. Same applies to other spec tests below.

File: config/initializers/new_framework_defaults.rb
Comment (nemodjuric) [inline config/initializers/new_framework_defaults.rb]: This is left intact to leave feedback application configured the same way as before. Once upgrade has happened, it's up to team who owns it to modify it.

File: config/cable.yml
Comment (nemodjuric) [inline config/cable.yml]: We don't use Action Cable, I commented out adapter configuration for production, just in case.

File: bin/setup
Comment (nemodjuric) [inline bin/setup]: Section above refers to RoadRunner initial setup used to create feedback app.

**Comment by sunil-sadasivan** [comment]
Awesome stuff @nemodjuric!

I'm just viewing this now--so you'll be testing this branch off of UAT before the merge, or has this been done?

I'm going to run feedback locally to ensure it works for me (I haven't run it before, so may take me a couple hours)

**Comment by nemodjuric** [comment]
@sunil-sadasivan Yes, in process of doing it. Will confirm here when the branch testing in UAT is done.

**Comment by sunil-sadasivan** [comment]
I'm a little hesitant to see the amount of major version gem upgrades with this change.  It makes sense that upgrading to Rails5 requires us to do this.  

For Caseflow and ex, I think we'll need to coordinate extensive testing in UAT by each sprint team to ensure all apps work as expected.  Does this make sense to you @amprokop @nemodjuric?  If so, once Caseflow is updated we can try to coordinate an effort with the other teams.

File: config/puma.rb
Comment (amprokop) [inline config/puma.rb]: did we not have a puma config file before?

**Review by amprokop — APPROVED** [review]
Two questions, LGTM when they're addressed.

**Comment by amprokop** [comment]
@nemodjuric — this specific Rails upgrade looks good to me after hearing that you successfully tested it in UAT. Can you make a separate issue with a rollout plan for other apps? When it comes time to upgrade caseflow-efolder/caseflow, we should pull in the tech leads from each team to inspect the changes and test in UAT themselves.

File: config/initializers/new_framework_defaults.rb
Comment (nemodjuric) [reply config/initializers/new_framework_defaults.rb]: This was auto-generated file by rails 5 update script. If you are referring to lines 22/23 - ActiveSupport.halt_callback_chains_on_return_false = true means to halt callback chains when a callback returns false, which is the old behavior.

File: config/initializers/session_store.rb
Comment (nemodjuric) [reply config/initializers/session_store.rb]: Update script had something else here, so I (accidentally) removed /n when I was editing it.

File: config/puma.rb
Comment (nemodjuric) [reply config/puma.rb]: We didn't. Puma became default web server in Rails 5.

**Pull request merged**
